test/gtest/uct/ib/test_rc.cc

```diff
@@ -134,6 +134,146 @@ UCS_TEST_P(test_rc_max_wr, send_limit)
+            UCS_TEST_ABORT("Error: cannot set get fc parameters");
+        }
+
+        if (UCS_OK != uct_config_modify(m_iface_config, "RC_TX_CQ_LEN", "32")) {
+            UCS_TEST_ABORT("Error: cannot modify RC_TX_CQ_LEN");
+        }
+
+        m_comp.count = 300000; // some big value to avoid func invocation
+        m_comp.func  = NULL;
+    }
+
+    unsigned gets_available(entity *e) {
+        return rc_iface(e)->tx.gets_available;
+    }
+
+protected:
+    unsigned         m_num_get_ops;
+    unsigned         m_max_get_zcopy;
+    uct_completion_t m_comp;
+};
+
+UCS_TEST_P(test_rc_get_fc, get_ops_limit)
+{
+    check_caps_skip(UCT_IFACE_FLAG_GET_ZCOPY | UCT_IFACE_FLAG_GET_BCOPY);
+
+    mapped_buffer sendbuf(1024, 0ul, *m_e1);
+    mapped_buffer recvbuf(1024, 0ul, *m_e2);
+
+    UCS_TEST_GET_BUFFER_IOV(iov, iovcnt, sendbuf.ptr(), sendbuf.length(),
+                            sendbuf.memh(), m_e1->iface_attr().cap.get.max_iov);
+
+    ucs_status_t status_b, status_z, status_exp;
+
+
+    for (unsigned i = 0; i <= m_num_get_ops;) {
+       status_z = uct_ep_get_zcopy(m_e1->ep(0), iov, iovcnt, recvbuf.addr(),
+                                  recvbuf.rkey(), &m_comp);
+       status_b = uct_ep_get_bcopy(m_e1->ep(0), (uct_unpack_callback_t)memcpy,
+                                   sendbuf.ptr(), sendbuf.length(),
+                                   recvbuf.addr(), recvbuf.rkey(), &m_comp);
+       i += 2;
+       status_exp = (i > m_num_get_ops) ? UCS_ERR_NO_RESOURCE : UCS_INPROGRESS;
+       EXPECT_EQ(status_exp, status_z);
+       EXPECT_EQ(status_exp, status_b);
+    }
+    EXPECT_EQ(0u, gets_available(m_e1));
+
+    // Check that it is possible to add to pending if get returns NO_RESOURCE
+    // due to lack of get credits
+    uct_pending_req_t pend_req;
+    pend_req.func = NULL; // Make valgrind happy
+    EXPECT_EQ(UCS_OK, uct_ep_pending_add(m_e1->ep(0), &pend_req, 0));
+    uct_ep_pending_purge(m_e1->ep(0), NULL, NULL);
+
+    flush();
+    EXPECT_EQ(m_num_get_ops, gets_available(m_e1));
+}
+
+// Check that get function fails for messages bigger than MAX_GET_ZCOPY value
+UCS_TEST_P(test_rc_get_fc, get_size_limit)
+{
+    check_caps_skip(UCT_IFACE_FLAG_GET_ZCOPY);
+
+    EXPECT_EQ(m_max_get_zcopy, m_e1->iface_attr().cap.get.max_zcopy);
+
+    mapped_buffer buf(m_max_get_zcopy + 1, 0ul, *m_e1);
+
+    UCS_TEST_GET_BUFFER_IOV(iov, iovcnt, buf.ptr(), buf.length(), buf.memh(),
+                            m_e1->iface_attr().cap.get.max_iov);
+
+    scoped_log_handler wrap_err(wrap_errors_logger);
+    ucs_status_t status = uct_ep_get_zcopy(m_e1->ep(0), iov, iovcnt,
+                                           buf.addr(), buf.rkey(), &m_comp);
+    EXPECT_EQ(UCS_ERR_INVALID_PARAM, status);
+
+    flush();
+    EXPECT_EQ(m_num_get_ops, gets_available(m_e1));
+}
+
+// Check that get size value is trimmed by the actual maximum IB msg size
+UCS_TEST_P(test_rc_get_fc, invalid_get_size)
+{
+    check_caps_skip(UCT_IFACE_FLAG_GET_ZCOPY);
```

Comment:
can we do this check inside UCS_TEST_P macro invocation?

Reply:
using `UCS_TEST_SKIP_COND_P` requires more code. I'll have to define `init` function in addition to adding checks, like 
```
UCS_TEST_SKIP_COND_P(test_rc_get_limit, get_ops_limit,
                                         !check_caps_skip(UCT_IFACE_FLAG_GET_ZCOPY |UCT_IFACE_FLAG_GET_BCOPY))
```

Reply:
but it would reduce testing time, so we should use UCS_TEST_SKIP_COND_P if possible, even if it means adding some more code

Reply:
the time would be nearly the same. It will fail somewhere in `init` after creating all needed uct instances. But ok, will change that

Reply:
unhandled?

Reply:
yeah, forgot in stash